Packs/MISP/Integrations/MISPV3/MISPV3.py

```diff
@@ -0,0 +1,1436 @@
+                                score=score, reliability=reliability, malicious_description="Match found in MISP")
+        indicator = get_dbot_indicator(dbot_type, dbot, value)
+        all_attributes = outputs.get('Attribute')
+        events_to_human_readable = get_events_related_to_scored_tag(all_attributes, found_tag)
+        attribute_highlights = reputation_command_to_human_readable(all_attributes, score, events_to_human_readable)
+        readable_output = tableToMarkdown(f'Results found in MISP for value: {value}', attribute_highlights,
+                                          removeNull=True)
+        readable_output += tableToMarkdown('Related events', list(found_related_events.values()))
+        return CommandResults(indicator=indicator,
+                              raw_response=misp_response,
+                              outputs=outputs,
+                              outputs_prefix='MISP',
+                              outputs_key_field='ID',
+                              readable_output=readable_output)
+    else:
+        dbot = Common.DBotScore(indicator=value, indicator_type=indicator_type, integration_name=INTEGRATION_NAME,
+                                score=Common.DBotScore.NONE, reliability=reliability,
+                                malicious_description="No results were found in MISP")
+        indicator = get_dbot_indicator(dbot_type, dbot, value)
+        return CommandResults(indicator=indicator,
+                              readable_output=f"No attributes found in MISP for value: {value}")
+
+
+def get_events_related_to_scored_tag(all_attributes, found_tag):
+    """
+    This function searches for all the events that have the tag (i.e found_tag) which caused the indicator to be scored
+    as malicious or suspicious.
+    Args:
+        all_attributes (dict): The parsed response from the MISP search attribute request
+        found_tag (str): The tag that was scored as malicious or suspicious. If no tag was found, then the score is
+        Unknown so no events should be found.
+
+    Returns:
+        list includes all the events that were detected as related to the tag.
+    """
+    related_events = []
+    if found_tag:
+        for attribute in all_attributes:
+            event = attribute.get('Event', {})
+            event_name = event.get('Info')
+            related_events.extend(search_events_with_scored_tag(event, found_tag, event_name))
+            related_events.extend(search_events_with_scored_tag(attribute, found_tag, event_name))
+    return remove_duplicated_related_events(related_events)
+
+
+def remove_duplicated_related_events(related_events):
+    related_events_no_duplicates = []
+    for i in range(len(related_events)):
+        if related_events[i] not in related_events[i + 1:]:
+            related_events_no_duplicates.append(related_events[i])
+    return related_events_no_duplicates
+
+
+def search_events_with_scored_tag(object_data_dict, found_tag, event_name):
```

Comment:
if you already have the event name and the tag when you call the function, why do we need to search it?